bootstrap/pkg/kfapp/coordinator/coordinator.go

```diff
@@ -142,7 +142,13 @@ func (coord *coordinator) getPackageManagers(kfdef *kfdefsv3.KfDef) *map[string]
 // kftypesv3.LoadKfApp which will try and dynamically load a .so
 //
 func getPackageManager(kfdef *kfdefsv3.KfDef) (kftypesv3.KfApp, error) {
-	switch kfdef.Spec.PackageManager {
+
+	packageManager := kfdef.Spec.PackageManager
```

Comment:
Shouldn't the defaults be filled in before we call getPackageManager? Specifically when we call NewKfApp
https://github.com/kubeflow/kubeflow/blob/2395c5fc0d384cf6dd4d7167bbe9010cd1315b54/bootstrap/pkg/kfapp/coordinator/coordinator.go#L468

Isn't that where any initialization code that sets default values should happen? So that's where we could set PackageManager to Kustomize.

Reply:
@jlewi `NewKfApp` isn't used anymore.
I don't think we have a clear way of doing defaulting right now.
Right now we have the following call chains:

build: `BuildKfAppFromURI` -> `NewLoadKfAppFromURI` which loads a KfDef (`LoadKFDefFromURI`), persists it as app.yaml (`CreateKfAppCfgFile`) and then loads it (`LoadKfAppCfgFile`).
apply: `LoadKfAppCfgFile`

IMO it makes most sense to have it inside `NewLoadKfAppFromURI`, I see that there are also some GCP checks in there.

Reply:
Doing it in the new functions seems better. Since we have multiple entry points I might suggest creating a new function "initDefaults" or something like that, that can be called from multiple entry points if needed.